values/table.go

```diff
@@ -0,0 +1,44 @@
+package values
+
+import "github.com/influxdata/flux/semantic"
+
+// TODO maybe a table should be indexed by columns directly.
+// A table is an object with a schema (object) and records (array of objects).
+var (
+	TableType = semantic.NewObjectPolyType(
+		map[string]semantic.PolyType{
+			"schema":  semantic.Object,
+			"records": semantic.NewArrayPolyType(semantic.Tvar(1)),
+		},
+		semantic.LabelSet{"schema", "records"},
+		semantic.LabelSet{"schema", "records"},
+	)
+	TableMonoType semantic.Type
+)
+
+func init() {
+	TableMonoType, _ = TableType.MonoType()
+}
+
+// Table is a values.Value that represent a table.
+// (Unlike flux.TableObject which represents a stream of tables.)
+type Table struct {
```

Comment:
It's probably a bad idea to copy this. If we're going to do things this way, I think it would be better to have this act as utility functions that operate on a `flux.Table` and implements the `values.Object` interface.

Also, this specific implementation is row-based, but all of our representations of a table are column-based.

Reply:
Yeah, totally agree with you. I am implementing with a wrapper for flux.Table right now.